syncweek-react-aad/react-aad
# Update dependencies to resolve security vulnerabilities, and enable new MSAL config options

Update NPM dependencies and enable all newer MSAL.js options to be utilized.

## Purpose
1) Using `npm audit` there are more than 300+ detected vulnerabilities. This is a big blocker for production apps or even publicly accessible POCs and demos.
2) MSAL.js has had a year of development, and there are many more options available when configuring the authentication. 

## Does this introduce a breaking change?
```
[ ] Yes
[ x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ x] Refactoring (no functional changes, no api changes)
[ x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code, perform `npm install` followed by `npm audit`. The result should show 0 vulnerabilities.
*  Walkthrough the sample app happy path to test usual behavior.

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```
## Other Information
- I noticed that before making any changes, the Sample app unit test is failing. I assume this has always been the case and seemed to be a test runner issue. I left this alone.
- The Sample app also has security vulnerabilities due to out-of-date dependency versions. I have not resolved these, as the sample project does not deploy with the npm package itself.
- This PR includes the same solution as PR #65 to enable a new MSAL config option.
- It's likely that with the newer version of MSAL.js there is additional functionality that this project should expose. I did not make an investigation to determine what the change delta is.
